src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.spec.ts

```diff
@@ -0,0 +1,37 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { HttpClient } from '@angular/common/http';
+import { FormsModule } from '@angular/forms';
+import { Router } from '@angular/router';
+import { AddNewSubredditComponent } from './add-new-subreddit.component';
+import { GatewayService } from 'src/app/services/gateway.service';
+
+class MockDataService {}
+
+class RouterTestingModule {}
```

Comment:
Why are you creating an empty router testing moduel if angular already provides one?

https://angular.io/api/router/testing/RouterTestingModule

note: It is not necessarily a mistake as you can mock everything yourself and for purposes of testing, it might just work.

Reply:
I do not need to use the static method with routes yet which is provided in the RouterTestingModule from angular. So I would like to leave it empty for now.